### DISCUSSION TITLE
Problem with OIDC login process with iTwin Sample Web App (FutureOn)

### DISCUSSION THREAD

**Discussion by PeterJBassett** [iTwin/itwinjs-core#1193] [Q&A] [answered]
We have built a sample iTwin Viewer react webapp. This works correctly when run on my Macbook (localhost). The OIDC login process is successful. However, when the web app is deployed to a development server, the OIDC login process fails with this error:

(index):1 Access to XMLHttpRequest at 'https://imsoidc.bentley.com/.well-known/openid-configuration' from origin 'https://futureon-itwin-viewer-frontendvue.toc.systems' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.

Please could you advise on the solution to this problem?

**Comment by aruniverse** [comment]
Has your development server been whitelisted (added to the list of allowed CORS origin and the redirect urls) in your oidc client config? If you setup your app using the [Getting Started](https://www.itwinjs.org/getting-started/registration-dashboard?tab=0) page, you can go to Dashboards > My Applications > Edit icon > and see list of all redirect uris currently configured for your app

**Comment by laurynasr** [comment, marked as answer]
It may take some time to replicate the configuration over different regions, and we've also seen some cases when the blocked response is being cached.
I have checked the configuration, the `https://futureon-itwin-viewer-frontendvue.toc.systems` origin is in the list of allowed origins; Can you try again and see if it's still not working for you?

**Reply by PeterJBassett** [reply]
Thank you for your reply, I have tried again, and it appears that the CORS issue has now gone.
The next step in the OIDC sequence is currently not working as my sample react itwin viewer application is setting the redirect_uri to localhost. I am not sure why at present, and am investigating the problem.

**Reply by laurynasr** [reply]
You need to change the `IMJS_AUTH_CLIENT_REDIRECT_URI` variable in `.env` file to contain URL for your deployment

**Reply by PeterJBassett** [reply]
Thanky ou for your reply and your help. Yes, this variable is set correctly in the .env file. So I am not sure why the redirect_uri is not correct.

**Reply by aruniverse** [reply]
Just to confirm, you aren't using the same `.env` file to point to both `localhost` and `https://futureon-itwin-viewer-frontendvue.toc.systems` right? If you used the itwin-viewer cra template, please check the `AuthorizationClient.ts` and make sure the value for the `redirectUri` in the oidcConfiguration object looks right and that uri exists in your app.
